Marlin/stepper.cpp

```diff
@@ -60,6 +60,55 @@
 
 Stepper stepper; // Singleton
 
+// public:
+
+block_t* Stepper::current_block = NULL;  // A pointer to the block currently being traced
+
+#if ENABLED(ABORT_ON_ENDSTOP_HIT_FEATURE_ENABLED)
+  bool Stepper::abort_on_endstop_hit = false;
+#endif
+
+#if ENABLED(Z_DUAL_ENDSTOPS)
+  bool Stepper::performing_homing = false;
+#endif
+
+// private:
+
+unsigned char Stepper::last_direction_bits = 0;        // The next stepping-bits to be output
+unsigned int Stepper::cleaning_buffer_counter = 0;
+
+#if ENABLED(Z_DUAL_ENDSTOPS)
+  bool Stepper::locked_z_motor = false;
+  bool Stepper::locked_z2_motor = false;
+#endif
+
+long  Stepper::counter_X = 0,
+      Stepper::counter_Y = 0,
+      Stepper::counter_Z = 0,
+      Stepper::counter_E = 0;
+
+volatile unsigned long Stepper::step_events_completed = 0; // The number of step events executed in the current block
+
+#if ENABLED(ADVANCE)
+  unsigned char Stepper::old_OCR0A;
+  long  Stepper::final_advance = 0,
+        Stepper::old_advance = 0,
+        Stepper::e_steps[4],
+        Stepper::advance_rate,
+        Stepper::advance;
+#endif
+
+long Stepper::acceleration_time, Stepper::deceleration_time;
+
+volatile long Stepper::count_position[NUM_AXIS] = { 0 };
+volatile signed char Stepper::count_direction[NUM_AXIS] = { 1 };
```

Comment:
I concern a little about this initializing.
If this initializing attempts that all elements of array are set by 1, it fails.
Only first element is set by 1, The others are set by 0.

sample:
```cpp
#define NUM_AXIS 4
#define EXTRUDERS 4

static volatile signed char count_direction[NUM_AXIS] = { 1 };
int autotune_temp[EXTRUDERS] = { 150 }; //in ultralcd.cpp

void setup() {
  Serial.begin(115200);
}

void loop() {
  Serial.println("count_direction");
  for (uint8_t i = 0; i < NUM_AXIS; i++) {
    Serial.print(count_direction[i]);
    if (i < NUM_AXIS - 1) Serial.print(",");
  }
  Serial.println("\n");
  delay(2000);

  Serial.println("autotune_temp");
  for (uint8_t i = 0; i < EXTRUDERS; i++) {
    Serial.print(autotune_temp[i]);
    if (i < NUM_AXIS - 1) Serial.print(",");
  }
  Serial.println("\n");
  delay(2000);
}
```